src/app/services/website/WebsiteDAO.ts

```diff
@@ -53,3 +53,32 @@ export function save(website : Website) : Promise <Website> {
 	emptyCheck(website);
 	return wrapSave(new WebsiteModel(website).save());
 }
+
+export function update(name : string, data:any) : Promise <Website> {
+	emptyCheck(name, data);
+	const {name: newName, hosts, chiefEditors, parserClass, layout} = data;
+	return WebsiteModel.findOne({ name })
+		.then(wsite => {
+			if (!wsite) {
+				throw new Error(`No such website ${name}`);
+			}
+			if (newName) {
+				wsite.name = newName;
+			}
+			if (hosts) {
+				wsite.hosts = hosts;
+			}
+			if (chiefEditors) {
+				wsite.chiefEditors = chiefEditors;
+			}
+			if (parserClass) {
+				wsite.parserClass = parserClass;
+			}
```

Comment:
my eyes bleed when I see all these `if(property){wsite.property=property}`
I have a suggestion if it works(AFAIR we have lodash loaded)
```
var data = _.pick(data, ['all', 'necessary', 'fields']) // take fields that we need
var filtered = _.pickBy(data) // remove empty
var resWsite = Object.assign(wsite, filtered); // merge them, not sure if that will manage with model instance, but worth of try
resWsite.save();
```
no pressure, just recommendation :bowtie: